module/modmanager/manager.go

```diff
@@ -604,6 +578,8 @@ func (mgr *Manager) attemptRestart(ctx context.Context, mod *module) []resource.
 		return orphanedResourceNames
 	}
 
+	mod.registerResources(mgr, mgr.logger)
```

Comment:
We were missing reregistration after restart of crashed modules 😮‍💨. I would not be surprised if there are more bugs lying around in here around reconfiguration, restarting, and the module manager's interactions with the resource manager.

Reply:
I won't be surprised either. It's a lot of very convoluted logic in here. But I think we just keep fixing and improving as we go and it'll shore things up soon enough.